packages/contractkit/src/wallets/signers/aws-hsm-signer.ts

```diff
@@ -0,0 +1,109 @@
+import { ensureLeading0x, publicKeyToAddress, trimLeading0x } from '@celo/utils/lib/address'
+import { KMS } from 'aws-sdk'
+import BigNumber from 'bignumber.js'
+import { ec as EC } from 'elliptic'
+import * as ethUtil from 'ethereumjs-util'
+import { parseBERSignature } from '../../utils/ber-utils'
+import { bigNumberToBuffer, bufferToBigNumber, isCanonical } from '../../utils/signature-utils'
+import { getHashFromEncoded, RLPEncodedTx } from '../../utils/signing-utils'
+import { Signer } from './signer'
+
+const SigningAlgorithm = 'ECDSA_SHA_256'
+const secp256k1Curve = new EC('secp256k1')
+
+function getRecoveryParam(message: Buffer, expectedAddress: string, r: Buffer, s: Buffer): number {
```

Comment:
Can we re-use `recoverKeyIndex` from [here](https://github.com/celo-org/celo-monorepo/blob/11814bdb3b44bddb3c19a9ec0f439ad45e86b6bc/packages/contractkit/src/utils/azure-key-vault-client.ts#L147)? From what I read it's possible for this key to be in the set [0,1,2,3], although 2,3 are very rare:
https://ethereum.stackexchange.com/questions/42455/during-ecdsa-signing-how-do-i-generate-the-recovery-id

Reply:
I've really struggled getting this to work... The ecdsaRecover fn I an using just calls into the `secp256k1` but I can't figure out what is going wrong.